backend/src/main/twirl/MainZIOHttp4s.scala.txt

```diff
@@ -0,0 +1,28 @@
+@(groupId: String)
+package @groupId
+
+import org.http4s.blaze.server.BlazeServerBuilder
+import org.http4s.server.Router
+import sttp.tapir.server.http4s.ztapir.ZHttp4sServerInterpreter
+import zio.interop.catz._
+import zio.{Scope, Task, ZIO, ZIOAppArgs, ZIOAppDefault}
+
+object Main extends ZIOAppDefault {
+
+  override def run: ZIO[Any with ZIOAppArgs with Scope, Any, Any] = {
+
+    val routes = ZHttp4sServerInterpreter().from(List(ApiDefinitions.helloServerEndpoint)).toRoutes
+
+    val serve: Task[Unit] =
+      BlazeServerBuilder[Task]
+        .withExecutionContext(runtime.runtimeConfig.executor.asExecutionContext)
+        .bindHttp(8080, "localhost")
+        .withHttpApp(Router("/" -> (routes)).orNotFound)
+        .serve
```

Comment:
w IO jest uzyte resource i  use a tu serve, mozna  w obu zrobic spojnie potencjalnie

Reply:
Good catch I will change ZIO to unify it: 

```
      BlazeServerBuilder[Task]
        .withExecutionContext(runtime.runtimeConfig.executor.asExecutionContext)
        .bindHttp(8080, "localhost")
        .withHttpApp(Router("/" -> (routes)).orNotFound)
        .resource
        .use { _ => ZIO.succeedBlocking(StdIn.readLine()) }
        .as(ExitCode.Success)
```